frontend/components/agent/AgentHeader.tsx

```
@@ -32,39 +35,65 @@ export function AgentHeader() {
           {/* Connection Status */}
-          <RealTimeStatus size="sm" />
+          <div className="hidden sm:block">
+            <RealTimeStatus size="sm" />
+          </div>
+
+          {/* Theme Toggle */}
+          <button
+            onClick={() => setTheme(theme === 'dark' ? 'light' : 'dark')}
+            className="p-2.5 text-muted-foreground hover:text-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 rounded-xl hover:bg-accent transition-all duration-200 micro-bounce"
+            title={`Switch to ${theme === 'dark' ? 'light' : 'dark'} mode`}
+          >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)